cbits/HsUnix.c

```diff
@@ -104,6 +104,15 @@ char *__hscore_d_name( struct dirent* d )
   return (d->d_name);
 }
 
+char __hscore_d_type( struct dirent* d )
+{
+#ifdef HAVE_DIRENT_D_TYPE
```

Comment:
Was this tested / did it work for you?

Apparently it is always false because the macro name is wrong:

https://github.com/haskell/unix/issues/347